clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -66,6 +70,56 @@ class CIRGenBuilderTy : public cir::CIRBaseBuilderTy {
                                     cir::ArrayType arrayTy) const {
     return cir::ConstArrayAttr::get(arrayTy, attrs);
   }
+  //
+  // Floating point specific helpers
```

Comment:
I think it's premature to introduce the skeleton of the constrained floating-point behavior. Let's omit all of this and just use a MissingFeatures marker where it would be called.